doc/user-manual/tools/interface-files.rst

```diff
@@ -0,0 +1,47 @@
+.. _interface-files:
+
+********************
+Interface files
+********************
+
+.. note::
+   This is a stub. Contributions, additions and corrections are greatly
+   appreciated.
+
+Probably you already noticed that when you save an agda file, another file
+with the same name and extension .agdai is automatically created. This .agdai
+file is what we call an **interface file** that stores a compressed form of
+the type-checked version of the code that lives in the corresponding .agda file.
+
+Interface files store the results from the type-checking process and resolve
+all implicit arguments (unless you use the flag ``--allow-unsolved-metas``).
+Formally, an .agdai file is a compiled version of the agda source code that can
+be executed by the Agda machine.
+
+Interface files are intended to make operations faster. They translate
```

Comment:
This paragraph is then obsolete.  (And it is slightly inaccurate, since the translation to case trees is already available before the interface file is created.  It is only that this translation is saved there permanently.)

Reply:
Updated